app-backend/tile/src/main/scala/LayerCache.scala

```diff
@@ -78,17 +82,41 @@ object LayerCache extends Config with LazyLogging {
         }
       }
     )
-  }
 
-  def layerHistogram(layerId: UUID, zoom: Int): OptionT[Future, Array[Histogram[Double]]] = {
+  def layerHistogram(layerId: UUID, zoom: Int): OptionT[Future, Array[Histogram[Double]]] =
     histogramCache.cachingOptionT(s"histogram-$layerId-$zoom") { implicit ec =>
       attributeStoreForLayer(layerId).map { case (store, _) => blocking {
         store.read[Array[Histogram[Double]]](LayerId(layerId.toString, 0), "histogram")
       }}
     }
-  }
 
-  def layerTile(layerId: UUID, zoom: Int, key: SpatialKey): OptionT[Future, MultibandTile] = {
+  def modelLayerHistogram(toolId: UUID, nodeId: Option[UUID]): OptionT[Future, Histogram[Double]] =
+    histogramCache.cachingOptionT(s"model-$toolId-$nodeId") { implicit ec =>
+      val futureTool: Future[Tool.WithRelated] =
+        Tools.getTool(toolId).map { _.getOrElse(throw new java.io.IOException(toolId.toString)) }
```

Comment:
Does this throw a `404` back to the user when there is no matching `tool` for a given `toolId`? That's what should be happening in this case right?